taprpc/universerpc/universe.proto

```diff
@@ -868,4 +876,49 @@ message FetchSupplyCommitResponse {
     // Each entry corresponds to the key at the same index in
     // `ignore_leaf_keys`.
     repeated bytes ignore_leaf_inclusion_proofs = 14;
+}
+
+message FetchSupplyLeavesRequest {
+    // The unique identifier for the target asset group whose supply leaves are
+    // being fetched.
+    oneof group_key {
+        // The 32-byte asset group key specified as raw bytes (gRPC only).
+        bytes group_key_bytes = 1;
+
+        // The 32-byte asset group key encoded as hex string (use this for
+        // REST).
+        string group_key_str = 2;
+    }
+
+    // The start block height for the range of supply leaves to fetch.
+    uint32 block_height_start = 3;
+
+    // The end block height for the range of supply leaves to fetch.
+    uint32 block_height_end = 4;
+}
+
+message SupplyLeafKey {
+    // The outpoint associated with the supply leaf.
+    Outpoint outpoint = 1;
+
+    // The script key of the supply leaf. This is the script key of the asset
+    bytes script_key = 2;
```

Comment:
So for the ignore tree, the key is actually the hash of the prev ID, which includes the asset ID.

Reply:
Similarly, we now use `AssetLeafKey` for the ignore and supply tree to fix prior issues related to collisions.

Reply:
This is important, as the client needs to reconstruct the tree exactly as the issuer created it.